interfaces/configfiles/spec.go

```diff
@@ -0,0 +1,102 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package configfiles
+
+import (
+	"fmt"
+
+	"github.com/snapcore/snapd/interfaces"
+	"github.com/snapcore/snapd/snap"
+)
+
+// Specification assists in collecting paths and content associated with an
+// interface.
+//
+// Unlike the Backend itself (which is stateless and non-persistent) this type
+// holds internal state that is used by the configfiles backend during the
+// interface setup process.
+type Specification struct {
+	// pathContent is the a map from absolute file paths to their expected content
+	pathContent map[string]string
+}
+
+// Methods called by interfaces
+
+// AddPathContent adds a configuration file with its content to the specification.
+func (spec *Specification) AddPathContent(path, content string) error {
+	if spec.pathContent == nil {
+		spec.pathContent = make(map[string]string)
+	}
+	if _, ok := spec.pathContent[path]; ok {
+		return fmt.Errorf("%s is already a managed configuration file", path)
+	}
+	spec.pathContent[path] = content
+	return nil
+}
+
+func (spec *Specification) PathContent() map[string]string {
+	return spec.pathContent
```

Comment:
Should this return a copy?

Reply:
This is used just for inspection in the tests, there is no need to do a copy. I have now moved it to export_test.go so that is clearer.